Tasks/Xcode/task.json

```diff
@@ -231,30 +231,24 @@
             "visibleRule": "signingOption = auto"
         },
         {
-            "name": "destinationPlatformOption",
+            "name": "destinationPlatform",
```

Comment:
I considered adding back `destinationPlatformOption` as an alias to make migrating from 4.* easier. But I expect issues if the build definition has:

```
 destinationPlatformOption: custom
 destinationPlatform: blah
```

Reply:
Ideally I could supply an "upgrade" script that transforms inputs when you move to 4.* to 5.*. I'll test to see how confusing it is to fix up a definition when moving from 4 to 5.

Reply:
Seems like even if you don't have to change the major/minor versions in this change, you should update the patch version.

Reply:
I don't think you should worry about any kind of upgrade script. Major version changes are not backwards compatible. I don't think we typically do any upgrade scripts.

Reply:
I tested moving from 4.* to 5.*, using the existing Xcode template as a starting point.

Xcode build: Just switch from 4.* to 5.*. The "Destination Platform" combobox goes blank. The build still succeeds.

Xcode test: Switch from 4.* to 5.*. The "Destination Platform" combobox goes blank. UI tests fail until you set "Destination Platform" back to "iOS and watchOS", the default you get from the Xcode template. The error is: 

```
xcodebuild: error: Failed to build workspace Xcode91WithDefaultTests with scheme Xcode91WithDefaultTests.
Reason: A build only device cannot be used to run this target.
```

It would be nice if moving from 4.* to 5.* usually just worked, but it's not too confusing. Maybe I can detect this error and add a warning that tells you to change destination platform.

Reply:
I'll use another PR for the improvement to add a warning.